### PR TITLE
ICMSLST-1919 Add support for cancel licenceData records.

### PR DESCRIPTION
**Pull request by MattHolmes123**

### CODE REVIEW

**Comment by MattHolmes123** [comment]
Here is an example of the file attachment it will generate for a revoke licence line:

image

**Comment by MattHolmes123** [comment]
Here is an example of the different payloads recieved:
image

File: mail/views.py [hunk 1]
```diff
@@ -67,12 +66,20 @@ def post(self, request):
...
-        icms_serializers = {
-            LicenceTypeEnum.IMPORT_OIL: serializers.FirearmOilLicenceDataSerializer,
-            LicenceTypeEnum.IMPORT_DFL: serializers.FirearmDflLicenceDataSerializer,
-            LicenceTypeEnum.IMPORT_SIL: serializers.FirearmSilLicenceDataSerializer,
-            LicenceTypeEnum.IMPORT_SAN: serializers.SanctionLicenceDataSerializer,
-        }
 
-        return icms_serializers[app_type]
+def get_serializer_cls(app_type: str, action: str) -> Type["Serializer"]:
+    match (app_type, action):
```
Comment (MattHolmes123) [inline mail/views.py:71]: I could change this to make it the same as in ICMS. (have a specific if block for the cancel serializer)

Comment (mikeylockwood) [reply mail/views.py:71]: I do think that is nicer than every match line having a `_` but not too concerned if you want to leave it as it is

**Review by mikeylockwood — APPROVED** [review]

File: mail/views.py (hunk 1 above)
Comment (MattHolmes123) [reply mail/views.py:71]: Yer I will update it.

**Pull request merged**